### PR TITLE
Update xeus package

### PR DESCRIPTION
**Pull request by tomstitt**
Support for `1.0.x` and `develop`

### CODE REVIEW

File: var/spack/repos/builtin/packages/xeus/package.py [hunk 1]
```diff
@@ -13,31 +15,53 @@ class Xeus(CMakePackage):
...
+
+    depends_on('xtl@0.4.0:', when='@:0.15.0')
+    depends_on('xtl@0.7.0:0.7.999', when='@1.0.4:,develop')
+
+    depends_on('nlohmann-json@3.4.0:', when='@1.0.4:,develop')
+    depends_on('nlohmann-json@3.2.0', when='@0.15.0')
     depends_on('nlohmann-json@3.1.1', when='@0.14.1')
-    depends_on('uuid')
+
+    depends_on('uuid', when=sys.platform == "linux")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/xeus/package.py]:
```suggestion
    depends_on('uuid', when='platform=linux')
```

File: var/spack/repos/builtin/packages/xeus/package.py [hunk 2]
```diff
@@ -13,31 +15,53 @@ class Xeus(CMakePackage):
...
+    maintainers = ['tomstitt']
+
     version('develop', branch='master')
+    version('1.0.4', sha256='7324ff013eb97d579fd3b6f9770a13f8863d6046c8bbcdbe2fc7d2ac02f0161f')
     version('0.15.0', sha256='bc99235b24d5757dc129f3ed531501fb0d0667913927ed39ee24281952649183')
     version('0.14.1', sha256='a6815845d4522ec279f142d3b4e92ef52cd80847b512146a65f256a77e058cfe')
 
     variant('examples', default=False, description="Build examples")
+    variant('shared', default=True, description="Build shared libraries")
+    variant('tests', default=False, description="Build tests")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/xeus/package.py]: Does this change anything in the binaries being produced or installed? If not I'd avoid a variant and rely only on `self.run_tests`

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/xeus/package.py (hunk 2 above)
Comment (tomstitt) [reply var/spack/repos/builtin/packages/xeus/package.py]: Yea, when tests are on there are some additional binaries being built

Comment (alalazo) [reply var/spack/repos/builtin/packages/xeus/package.py]: Are those binaries installed too?

Comment (tomstitt) [reply var/spack/repos/builtin/packages/xeus/package.py]: No they are not installed as far as I can tell. How does `self.run_tests` work (is it just a bool)? Something like:

```
...
args.append(self.define('XEUS_BUILD_TESTS', self.enable_tests))
...
```

?

Comment (alalazo) [reply var/spack/repos/builtin/packages/xeus/package.py]: Yes, something like:
```python
args.append(self.define('XEUS_BUILD_TESTS', self.run_tests))
```
should do it.

File: var/spack/repos/builtin/packages/xeus/package.py [hunk 3]
```diff
@@ -13,31 +13,53 @@ class Xeus(CMakePackage):
...
+
+    depends_on('xtl@0.4.0:', when='@:0.15.0')
+    depends_on('xtl@0.7.0:0.7.999', when='@1.0.4:,develop')
+
+    depends_on('nlohmann-json@3.4.0:', when='@1.0.4:,develop')
+    depends_on('nlohmann-json@3.2.0', when='@0.15.0')
     depends_on('nlohmann-json@3.1.1', when='@0.14.1')
-    depends_on('uuid')
+
+    depends_on('uuid', when='platform=linux')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/xeus/package.py:45]: Is this not needed on macOS? We added a `apple-libuuid` package and a `uuid` virtual dependency specifically to handle this kind of things.

File: var/spack/repos/builtin/packages/xeus/package.py [hunk 4]
```diff
@@ -13,31 +13,53 @@ class Xeus(CMakePackage):
...
     def cmake_args(self):
-        args = [
-            self.define_from_variant('BUILD_EXAMPLES', 'examples')
-        ]
+        args = []
+
+        if "@:0.15.0" in self.spec:
+            args.append(self.define_from_variant('BUILD_EXAMPLES', 'examples'))
+
+        elif "@1.0.4:" in self.spec or "@develop" in self.spec:
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/xeus/package.py]:
```suggestion
        elif "@1.0.4:" in self.spec:
```
`@develop` is included in `@1.0.4:`

File: var/spack/repos/builtin/packages/xeus/package.py [hunk 5]
```diff
@@ -13,31 +13,53 @@ class Xeus(CMakePackage):
...
     conflicts('%gcc@:4.8')
     conflicts('%clang@:3.6')
     conflicts('%intel@:17')
 
     depends_on('libzmq@4.2.5:-libsodium')
-    depends_on('cppzmq@4.3.0:')
-    depends_on('cryptopp@7.0.0:')
-    depends_on('xtl@0.4.0:')
-    depends_on('nlohmann-json@3.2.0', when='@develop,0.15.0:')
+    depends_on('cppzmq@4.7.1:', when='@1.0.4:,develop')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/xeus/package.py]:
```suggestion
    depends_on('cppzmq@4.7.1:', when='@1.0.4:')
```

File: var/spack/repos/builtin/packages/xeus/package.py [hunk 6]
```diff
@@ -13,31 +13,53 @@ class Xeus(CMakePackage):
...
+    depends_on('cppzmq@4.3.0:', when='@:0.15.0')
+
+    depends_on('cryptopp@7.0.0:', when='@:0.15.0')
+
+    depends_on('openssl@1.0.1:', when='@1.0.4:,develop')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/xeus/package.py]:
```suggestion
    depends_on('openssl@1.0.1:', when='@1.0.4:')
```

File: var/spack/repos/builtin/packages/xeus/package.py [hunk 7]
```diff
@@ -13,31 +13,53 @@ class Xeus(CMakePackage):
...
+
+    depends_on('xtl@0.4.0:', when='@:0.15.0')
+    depends_on('xtl@0.7.0:0.7.999', when='@1.0.4:,develop')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/xeus/package.py]:
```suggestion
    depends_on('xtl@0.7.0:0.7.999', when='@1.0.4:')
```

File: var/spack/repos/builtin/packages/xeus/package.py [hunk 8]
```diff
@@ -13,31 +13,53 @@ class Xeus(CMakePackage):
...
+
+    depends_on('nlohmann-json@3.4.0:', when='@1.0.4:,develop')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/xeus/package.py]:
```suggestion
    depends_on('nlohmann-json@3.4.0:', when='@1.0.4:')
```

File: var/spack/repos/builtin/packages/xeus/package.py (hunk 3 above)
Comment (tomstitt) [reply var/spack/repos/builtin/packages/xeus/package.py:45]: I did this based on their README: https://github.com/jupyter-xeus/xeus

Specifically, under [Building from Source](https://github.com/jupyter-xeus/xeus#building-from-source) they list their dependencies then say:
> On Linux platforms, xeus also requires `libuuid`, which is available in all linux distributions (`uuid-dev` on Debian).

**Review by alalazo — APPROVED** [review]

**Pull request merged**